scripts/pyenv_setup.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+# This script correctly sets up pyenv
+#
+# Assumptions:
+# - This script assumes you're calling from the top directory of the repository
+set -eu
+
+# Check if a command is available
+require() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+query_big_sur() {
+    if require sw_vers && sw_vers -productVersion | grep -E "11\." > /dev/null; then
+        return 0
+    fi
+    return 1
+}
+
+get_shell_startup_script() {
+  local _startup_script=''
+  if [ -n "$SHELL" ]; then
+    case "$SHELL" in
+      */bash)
+        _startup_script="${HOME}/.bash_profile"
```

Comment:
profiles and rcs are different things. In my experience, it's better to just put everything into the rcs. Don't ask, just.... trust me on this one.

Reply:
On GHA unless I put it on `.bash_profile`, it won't work since the order of loading is: load .bash_profile -> on 1st line load .bashrc -> continue modifying a long list of things on .bash_profile that make the path modifications of `pyenv init -` get lost.

Pyenv's recommendation is to put it in `.bash_profile`

Also from bash's man page:
> When bash is invoked as an interactive login shell, or as a non-interactive shell with the --login option, it first reads and executes commands from the file /etc/profile, if that file exists.  After reading that file, it looks for  ~/.bash_profile,  ~/.bash_login,  and ~/.profile, in that order, and reads and executes commands from the first one that exists and is readable.

I would like to stick with the official recommended practices.

Reply:
Hmm, okay.